### PR TITLE
Fixes #10337 - added ability to set the out of sync duration

### PR DESCRIPTION
**Pull request by ab5w**

### CODE REVIEW

File: app/models/setting/puppet.rb [hunk 1]
```diff
@@ -6,6 +6,7 @@ def self.load_defaults
     self.transaction do
       [
         self.set('puppet_interval', N_("Puppet interval in minutes"), 30 ),
+        self.set('outofsync_interval', N_("Duration in minutes after the puppet interval for servers to be classed as out of sync."), 5 ),
```
Comment (domcleal) [inline app/models/setting/puppet.rb]: tiny nitpick, please capitalise "Puppet"

**Comment by domcleal** [comment]
[test]

**Comment by domcleal** [comment]
@ab5w many of our tests are failing as I think it's missing in the test environment.  Could you try adding the new setting to test/fixtures/settings.yml?

Please squash the commits and push back to this pull request.  Thanks!

File: app/models/host/managed.rb [hunk 2]
```diff
@@ -65,7 +65,7 @@ class Jail < ::Safemode::Jail
   attr_reader :cached_host_params
 
   scope :recent,      lambda { |*args| {:conditions => ["last_report > ?", (args.first || (Setting[:puppet_interval] + 5).minutes.ago)]} }
```
Comment (domcleal) [inline app/models/host/managed.rb:67]: Should the '5' here be replaced too?

File: app/controllers/hosts_controller.rb [hunk 3]
```diff
@@ -504,8 +504,8 @@ def pending
   end
```
Comment (domcleal) [inline app/controllers/hosts_controller.rb:504]: The methods above here also have '5' referenced, update these too?

File: app/views/dashboard/_status_widget.html.erb [hunk 4]
```diff
@@ -23,7 +23,7 @@
   %>
 
   <%= searchable_links _('Out of sync hosts'),
```
Comment (domcleal) [inline app/views/dashboard/_status_widget.html.erb:25]: There are some other searches above which reference '5' still.

Comment (ab5w) [inline app/views/dashboard/_status_widget.html.erb:25]: They appear to be for different searches though, this request is just for the out of sync one.

Comment (domcleal) [inline app/views/dashboard/_status_widget.html.erb:25]: Though the calculation is perhaps for the same thing - to only consider hosts that are in sync and some other state?

**Comment by dLobatog** [comment]
[test]

**Comment by dLobatog** [comment]
Merged as 81884922f1a2849d98cbd4b82a26cac5e34b4158 , thanks @ab5w, and @domcleal for reviewing!
I squashed your commits into one, please remember to do so next time :smile_cat: 

The :recent search scope is OK with 5.minutes.ago as far as I understand it, it doesn't say anywhere it means 'not out of sync hosts', just recently updated ones.

**Pull request closed without merging**